oauthlib/oauth1/rfc5849/signature.py

```diff
@@ -199,77 +226,29 @@ def base_string_uri(uri, host=None):
     return v.replace(' ', '%20')
 
 
-# ** Request Parameters **
-#
-#    Per `section 3.4.1.3`_ of the spec.
-#
-#    In order to guarantee a consistent and reproducible representation of
-#    the request parameters, the parameters are collected and decoded to
-#    their original decoded form.  They are then sorted and encoded in a
-#    particular manner that is often different from their original
-#    encoding scheme, and concatenated into a single string.
-#
-# .. _`section 3.4.1.3`: https://tools.ietf.org/html/rfc5849#section-3.4.1.3
-
-def collect_parameters(uri_query='', body=[], headers=None,
+def collect_parameters(uri_query='', body=None, headers=None,
                        exclude_oauth_signature=True, with_realm=False):
-    """**Parameter Sources**
+    """
+    Gather the request parameters from all the parameter sources.
+
+    This function is used to extract all the parameters, which are then passed
+    to ``normalize_parameters`` to produce one of the components that make up
+    the *signature base string*.
 
     Parameters starting with `oauth_` will be unescaped.
 
     Body parameters must be supplied as a dict, a list of 2-tuples, or a
-    formencoded query string.
+    form encoded query string.
 
     Headers must be supplied as a dict.
 
-    Per `section 3.4.1.3.1`_ of the spec.
```

Comment:
Why did you remove the example?

Reply:
It added little value, because:
1. The reader can easily load up the RFC and read it directly.
2. The inline comments for the algorithm are already present.
3. I doubt readers will scroll up and down between the code and example. And if they did, it would be easier done in a separate window (point 1).
3. If the whole example was to be added as a comment, it would make more sense to copy it into the unit test _tests/oauth1/rfc5849/test_signatures.py_ which uses the example for its testing. But again, duplicating the RFC verbatim doesn't add much value.